nativelib/src/main/scala/java/lang/ClassInstancesRegistry.scala

```diff
@@ -0,0 +1,36 @@
+package java.lang
+
+/** Registry for created instances of java.lang.Class
+ * It's only purpose is to prevent GC from collecting instances of java.lang.Class
+ **/
+object ClassInstancesRegistry {
```

Comment:
You can make this `private[lang]` to make it clear that this is not a public API:
```suggestion
private[lang] object ClassInstancesRegistry {
```